content/shmem_team_get_config.tex

```diff
@@ -5,18 +5,21 @@
 \begin{apidefinition}
 
 \begin{Csynopsis}
-int @\FuncDecl{shmem\_team\_get\_config}@(shmem_team_t team, shmem_team_config_t *config);
+int @\FuncDecl{shmem\_team\_get\_config}@(shmem_team_t team, long config_mask, shmem_team_config_t *config);
 \end{Csynopsis}
 
 \begin{apiarguments}
   \apiargument{IN}{team}{An \openshmem team handle.}
+  \apiargument{IN}{config\_mask}{
+    The bitwise mask representing the set of configuration parameters to fetch from the given team.
+    }
   \apiargument{OUT}{config}{
     A pointer to the configuration parameters for the given team.}
 \end{apiarguments}
 
 \apidescription{
 \FUNC{shmem\_team\_get\_config} returns through the \VAR{config} argument
-the configuration parameters of the given team, which were assigned according
+the configuration parameters as described by the mask, which were assigned according
 to input configuration parameters when the team was created.
```

Comment:
Do we allow the team parameters to change over time? Would it provide more room for implementations if we wrote something like `the current parameters of \ARG{team}, as described by \ARG{config_mask}`?

Reply:
I'm not sure about changing configuration over the time. I can be useful to give an option to the implementation to push-back on some of the configuration and let's say allocate smaller number of resources. This happens during team creation. I don't expect this to change over time.

Reply:
Ignoring my earlier comment about parameters being dynamic, what do you think of:
> \FUNC{shmem\_team\_get\_config} returns through the \VAR{config} argument
the current values of the configuration parameters as described by the mask.

Reply:
What do you mean by ```current values```? For some implementation, there might not be any current value. So, should those implementations return 0 for ```num_contexts```? I think it is better to leave the values as asked by the users and not report anything modified internally by the implementation.

Reply:
We had a discussion in the Teams WG for https://github.com/gmegan/specification/issues/120 and decided to return the actual config passed by the users and not report the current/implementation value.

Reply:
@naveen-rn We had that semantic in MPI for years and just went through a long process to change it to get-info routines returning the current values in the [latest draft](https://www.mpi-forum.org/docs/drafts/mpi-2019-draft-report.pdf). This was done so that the implementation can give feedback to the application. For example, with `MPI_Comm_set_info` the user can check whether any of the new info values were accepted by the implementation.

I think your `num_contexts` example is really pointing out a gap in the specification, where we didn't provide an "unspecified"  value for this option.

Reply:
So, the question now if we want to stick with existing semantics, as for now and work on potential improvement for 1.6, including the gaps. Even if we decide to change this to "current" it is not an API change and I don't believe it will have any impact on applications.

Reply:
Given that there is only one field specified, it may not be a big deal from the spec point-of-view to delay to OpenSHMEM 1.6. However, implementations may also define their own fields in the meantime. It sounds like we don't have enough support in the WG to resolve this now, so I will file an issue.

Reply:
Ok, another question on this -- What value should the user get if they query `num_contexts` but did not set it when the team was created?

Reply:
I guess this should return `0`, which is the default value.

Sub-question. Why is the `num_contexts` field an `int` and not `size_t`? I don't think we specify the behavior if a negative number is provided.

Reply:
The advantage of int that in a future we can use negative values for some pre-defined values, a.k.a it is much easier to extend  in a future.

Reply:
Sorry, there were two comments in there -- why is it 32-bits and why is it signed. We could use `ssize_t` or `int64_t` but we still should clarify that negative values are invalid.